nucypher/blockchain/eth/sol/source/contracts/staking_contracts/WorkLockPoolingContract.sol

```diff
@@ -0,0 +1,439 @@
+        token.safeTransferFrom(msg.sender, address(this), _value);
+        emit TokensDeposited(msg.sender, _value, delegator.depositedTokens);
+    }
+
+    /**
+     * @notice Delegator can transfer ETH directly to workLock
+     */
+    function escrowETH() external payable {
+        Delegator storage delegator = delegators[msg.sender];
+        delegator.depositedETHWorkLock = delegator.depositedETHWorkLock.add(msg.value);
+        totalWorkLockETHReceived = totalWorkLockETHReceived.add(msg.value);
+        workLock.bid{value: msg.value}();
+        emit Bid(msg.sender, msg.value);
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function bid(uint256) public payable {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function withdrawCompensation() public pure {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function cancelBid() public pure {
+        revert();
+    }
+
+    /**
+     * @dev Hide method from StakingInterface
+     */
+    function claim() public pure {
+        revert();
+    }
+
+    /**
+     * @notice Claim tokens in WorkLock and save number of claimed tokens
+     */
+    function claimTokensFromWorkLock() public {
+        workLockClaimedTokens = workLock.claim();
+        totalDepositedTokens = totalDepositedTokens.add(workLockClaimedTokens);
+        emit Claimed(address(this), workLockClaimedTokens);
+    }
+
+    /**
+     * @notice Calculate and save number of claimed tokens for specified delegator
+     */
+    function calculateAndSaveTokensAmount() external {
+        Delegator storage delegator = delegators[msg.sender];
+        calculateAndSaveTokensAmount(delegator);
+    }
+
+    /**
+     * @notice Calculate and save number of claimed tokens for specified delegator
+     */
+    function calculateAndSaveTokensAmount(Delegator storage _delegator) internal {
+        if (workLockClaimedTokens == 0 ||
+            _delegator.depositedETHWorkLock == 0 ||
+            _delegator.claimedWorkLockTokens)
+        {
+            return;
+        }
+
+        uint256 delegatorTokensShare = _delegator.depositedETHWorkLock.mul(workLockClaimedTokens)
+            .div(totalWorkLockETHReceived);
+
+        _delegator.depositedTokens = _delegator.depositedTokens.add(delegatorTokensShare);
+        _delegator.claimedWorkLockTokens = true;
+        emit Claimed(msg.sender, delegatorTokensShare);
+    }
+
+    /**
+     * @notice Get available reward for all delegators and owner
+     */
+    function getAvailableReward() public view returns (uint256) {
+        uint256 stakedTokens = escrow.getAllTokens(address(this));
+        uint256 freeTokens = token.balanceOf(address(this));
+        uint256 reward = stakedTokens.add(freeTokens).sub(totalDepositedTokens);
+        if (reward > freeTokens) {
+            return freeTokens;
+        }
+        return reward;
```

Comment:
I'm having a hard time understanding the logic here.

We have `reward = staked + free - total`
Then we have two cases: 
1) if `reward > free` (which is equivalent to `staked + free - total > free`, and hence to `staked > total`, see comment above), we return `free`
2) otherwise, we return `reward`.

So essentially, when we have more tokens staked than deposited, the available rewards are coming exclusively from the pool balance (?); otherwise, it's the pool balance + rewards in StakingEscrow.  I'm afraid I don't understand.

Reply:
Looks like you said everything correct. Reward is everything that more than deposited, so we get info about all tokens which belonging to pool and then subtract totalDeposited. And we can't give reward more than contract has at contract address